code/modules/events/crystal_event.dm

```diff
@@ -254,30 +274,28 @@ This section is for the event controller
 
 /turf/open/indestructible/crystal_floor
 	name = "Crystal floor"
-	desc = "A crystalyzed floor"
+	desc = "A Crystallized floor"
 	icon_state = "noslip-damaged1"
 	baseturfs = /turf/open/space
 
 /turf/open/indestructible/crystal_floor/examine(mob/user)
 	. += ..()
 	. += "<span class='notice'>The floor is made of sturdy crystals.</span>"
 
-/turf/open/indestructible/crystal_floor/attackby(obj/item/W, mob/user, params)
-	if(W.tool_behaviour == TOOL_WELDER)
-		if(!W.tool_start_check(user, amount=0))
-			return FALSE
-
-		to_chat(user, "<span class='notice'>You begin heating up the crystal...</span>")
-		if(W.use_tool(src, user, 2.5 SECONDS, volume=100))
-			if(iswallturf(src))
-				to_chat(user, "<span class='notice'>The crystal crumbles into dust.</span>")
-				ScrapeAway(flags = CHANGETURF_INHERIT_AIR)
-			return TRUE
-	return FALSE
+/turf/open/indestructible/crystal_floor/welder_act(mob/living/user, obj/item/I)
+	. = ..()
+	if(!I.tool_start_check(user, amount=0))
+		return FALSE
+	to_chat(user, "<span class='notice'>You begin heating up the crystal...</span>")
+	if(I.use_tool(src, user, 2.5 SECONDS, volume=100))
+		if(iswallturf(src))
+			to_chat(user, "<span class='notice'>The crystal crumbles into dust.</span>")
+			ScrapeAway(flags = CHANGETURF_INHERIT_AIR)
+		return TRUE
```

Comment:
* Avoid `istype(src)` checks. Make redefined procs instead.
* Are you sure you want to return `TRUE` only if the `use_tool()` timed action succeeds? Wont it bash the floor else?

Reply:
`Avoid istype(src) checks. Make redefined procs instead.` how do i do this?

Reply:
hold on a moment, this check cannot ever be true
it's called by `/turf/open/indestructible/crystal_floor` and it's checking if the source is `/turf/closed/wall` !

Reply:
Oops copy paste error always gets me lol. Also maybe I don't even need it

Reply:
Definitely unnecessary.